apps/api/src/services/competition.service.ts

```diff
@@ -2265,6 +2267,84 @@ export class CompetitionService {
     }
   }
 
+  /**
+   * Check and automatically start competitions that have reached their start date
+   * Conditions:
+   * - No other competition is currently active
+   * - Competition is not in sandbox mode
+   * - Competition has at least one registered agent
+   * - Process competitions by earliest startDate first
+   */
+  async processCompetitionStartDateChecks(): Promise<void> {
+    try {
+      // Do not start anything if there's already an active competition
+      const active = await findActive();
+      if (active) {
+        serviceLogger.debug(
+          {
+            competitionId: active.id,
+            name: active.name,
+          },
+          `[CompetitionManager] Active competition found. Skipping auto-start checks`,
+        );
+        return;
+      }
+
+      const competitionsToStart = await findCompetitionsNeedingStarting();
+      if (competitionsToStart.length === 0) {
+        serviceLogger.debug(
+          "[CompetitionManager] No competitions ready to start",
+        );
+        return;
+      }
+
+      // We only support running one competition at a time, so we will not start any competitions
+      // if we find more than one. Note: This should not happen if competitions are created with
+      // the correct start dates; it's defensive.
+      const competition = competitionsToStart[0];
+      if (competitionsToStart.length > 1 || !competition) {
+        serviceLogger.warn(
+          {
+            competitions: competitionsToStart.map((c) => ({
+              id: c.id,
+              name: c.name,
+              startDate: c.startDate?.toISOString(),
+            })),
+          },
+          `[CompetitionManager] Multiple competitions ready to start. Skipping auto-start checks`,
+        );
+        return;
+      }
+      serviceLogger.debug(
+        {
+          competitionId: competition.id,
+          name: competition.name,
+          startDate: competition.startDate?.toISOString(),
+        },
+        `[CompetitionManager] Auto-starting competition`,
+      );
+      await this.startCompetition(competition.id);
+      serviceLogger.debug(
+        {
+          competitionId: competition.id,
+          name: competition.name,
+        },
+        `[CompetitionManager] Successfully auto-started competition`,
+      );
+    } catch (error) {
+      // Continue silently if the competition has no registered nor provided agents
```

Comment:
i _think_ the silent failure is how we want to handle it here? we outsource the "check if there are any agents in this comp" to the `startCompetition` method, which throws a 400 if there are no registered agent.

it saves us a query.